jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -41,6 +43,8 @@
   /** Helper for loading a .class file. */
   private static class ClassFileLoader extends ClassLoader {
 
+    private static final Map<Path, Class> cache = new HashMap<>();
```

Comment:
`Class<?>` instead of `Class` (same for `findClass` return type)

Reply:
Can't change the `findClass` return type since it's an override, but I'll change this one.

Reply:
Isn't the return type of `ClassLoader#findClass` `Class<?>`?

https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#findClass-java.lang.String-